.github/workflows/test-action.yml

```diff
@@ -0,0 +1,344 @@
+name: PR testing of CLI action
+
+on:
+  pull_request:
+    types: [ opened, synchronize, reopened, ready_for_review ]
+
+jobs:
+  should-workflow-run:
+    runs-on: ubuntu-latest
+    steps:
+      - if: >
+          !github.event.pull_request.draft && !(
+            (github.actor == 'asyncapi-bot' && (
+              startsWith(github.event.pull_request.title, 'ci: update of files from global .github repo') || 
+              startsWith(github.event.pull_request.title, 'chore(release):')
+            )) ||
+            (github.actor == 'asyncapi-bot-eve' && (
+              startsWith(github.event.pull_request.title, 'ci: update of files from global .github repo') || 
+              startsWith(github.event.pull_request.title, 'chore(release):')
+            )) ||
+            (github.actor == 'allcontributors[bot]' && 
+              startsWith(github.event.pull_request.title, 'docs: add')
+            )
+          )
+        id: should_run
+        name: Should Run
+        run: echo "shouldrun=true" >> $GITHUB_OUTPUT
+    outputs:
+      shouldrun: ${{ steps.should_run.outputs.shouldrun }}
+
+  build-docker:
+    needs: should-workflow-run
+    name: Build Docker image
+    if: ${{ needs.should-workflow-run.outputs.shouldrun == 'true' }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Get docker version
+        id: docker_version
+        run: >
+          ls -la;
+          action=$(cat action.yml);
+          regex='docker:\/\/asyncapi\/github-action-for-cli:([0-9.]+)';
+          [[ $action =~ $regex ]];
+          action_version=${BASH_REMATCH[1]};
+          echo "action_version=$action_version" >> $GITHUB_OUTPUT
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+      - name: Build Docker image and export
+        uses: docker/build-push-action@v5
+        with:
+          context: .
```

Comment:
The context should be different I believe as the DockerFile is inside github-action folder.